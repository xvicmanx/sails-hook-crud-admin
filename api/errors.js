const missingParamError = (param) => ({
  error: {
    message: `'${param}' missing!`,
    fields: {
      [param]: `'${param}' missing!`,
    },
  },
});

const userOrPassError = () => ({
  error: {
    message: 'Invalid "username or password"',
    fields: {
      username: 'Invalid "username or password"',
      password: 'Invalid "username or password"',
    }
  }
});

const systemError = () => ({
  error: {
    message: 'Something happened while authenticating :(',
  },
});

const tokenAuthError = () => ({
  error: {
    message: `Error in jwt token`,
    fields: {
      'jwt-token': `Error in jwt token`,
    },
  },
});

module.exports = {
  userOrPassError,
  systemError,
  missingParamError,
  tokenAuthError,
};
