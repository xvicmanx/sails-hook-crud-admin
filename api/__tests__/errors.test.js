import errors from '../errors';

describe('api::errors', () => {
  describe('missingParamError', () => {
    test('it returns the correct object', () => {
      expect(errors.missingParamError('test-param')).toMatchSnapshot();
    });
  });

  describe('userOrPassError', () => {
    test('it returns the correct object', () => {
      expect(errors.userOrPassError()).toMatchSnapshot();
    });
  });

  describe('systemError', () => {
    test('it returns the correct object', () => {
      expect(errors.systemError()).toMatchSnapshot();
    });
  });

  describe('tokenAuthError', () => {
    test('it returns the correct object', () => {
      expect(errors.tokenAuthError()).toMatchSnapshot();
    });
  });
});
