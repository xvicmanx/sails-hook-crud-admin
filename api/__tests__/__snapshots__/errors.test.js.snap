// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api::errors missingParamError it returns the correct object 1`] = `
Object {
  "error": Object {
    "fields": Object {
      "test-param": "'test-param' missing!",
    },
    "message": "'test-param' missing!",
  },
}
`;

exports[`api::errors systemError it returns the correct object 1`] = `
Object {
  "error": Object {
    "message": "Something happened while authenticating :(",
  },
}
`;

exports[`api::errors tokenAuthError it returns the correct object 1`] = `
Object {
  "error": Object {
    "fields": Object {
      "jwt-token": "Error in jwt token",
    },
    "message": "Error in jwt token",
  },
}
`;

exports[`api::errors userOrPassError it returns the correct object 1`] = `
Object {
  "error": Object {
    "fields": Object {
      "password": "Invalid \\"username or password\\"",
      "username": "Invalid \\"username or password\\"",
    },
    "message": "Invalid \\"username or password\\"",
  },
}
`;
